contributors/design-proposals/add-metadata-indicating-to-replace-union-in-patch.md

```diff
@@ -1,78 +1,61 @@
 Add Tag Indicating To Replace *Union* Fields in Patch (a.k.a Support Replacing Maps in Strategic Merge Patch)
 =============
 
-This approach could be used to fix the issues with `kubectl apply` simply by having apply always replace union fields instead of patching them.  E.g. instead of setting `spec.strategy.type` without touching the rest of `spec.strategy`, patches should always replace the entire `spec.strategy`.
+This approach could be used to fix the issues with `kubectl apply` simply by having apply always replace union fields instead of patching them.
 
 ## Proposed Changes
 
 ### APIs
 
-Add a go struct tag/openAPI value telling clients to replace the entire struct when doing a *PATCH*. For the *inlined* fields, we have 2 options:
+Add a go struct tag / openAPI value telling clients to replace the entire struct when doing a `PATCH`. For the *inlined* fields, we have 2 options:
 
-1) Move the tag to the parent struct's. e.g. add a tag to `Volume` instead of `VolumeSource`. 
-This will limit the flexibility to add more fields in the parent struct that already has an inlined union. 
-And we need to examine all the impacted APIs to check if it is safe to replace the entire parent struct. e.g. if some controller is updating some fields.
+1) Move the tag to the parent struct's. And we replace the entire parent struct.
 
-Examples of how to add go struct tag for option 1:
-```go
-type PodSpec struct {
-	...
-	// patchStrategyForListEntry:"replace" is an additional struct tag indicating we replace the entire entry in the list, if they have the same merge key. 
-	Volumes []Volume `json:"volumes,omitempty" patchStrategy:"merge" patchMergeKey:"name" patchStrategyForListEntry:"replace" protobuf:"bytes,1,rep,name=volumes"`
-	...
-}
-type Volume struct {
-	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
-	VolumeSource `json:",inline" protobuf:"bytes,2,opt,name=volumeSource"`
-}
+We reuse `patchStrategy:"replace"` if possible. This kind of patch will be backward compatible.
+
+Otherwise, we introduce a new struct tag, `patchStrategyForListEntry:"replace"`, for a list of inlined unions. This kind of patch will not be backward compatible.
```

Comment:
isn't the point of a union type to ensure only one of a set of fields is set *at that level* in the object? both replace tags seem to remove the possibility of merging further down in the tree if the specified member of the union type didn't change.